index.html

```diff
@@ -202,82 +181,60 @@ <h1>Prototype Viewer MediaWiki 3D GLTF Data, STL data is already available </h1>
-            if (document.getElementById('container')) {
-
-                this.scene.clear();
-                const child = document.getElementById('container');
-                document.body.removeChild(child);
-            }
-            container = document.createElement('div');
-            container.id = 'container';
-            document.body.appendChild(container);
-            const dimensions = this.getDimensions();
-            // renderer
-            this.renderer = new THREE.WebGLRenderer({ antialias: true });
-            // this.renderer.setClearColor(0x222222);
-            this.renderer.setPixelRatio(window.devicePixelRatio);
-            this.renderer.setSize(dimensions.width, dimensions.height);
-            this.renderer.shadowMap.enabled = true;
-            document.getElementById('container').appendChild(this.renderer.domElement);
-            // camera
-            this.camera = new THREE.PerspectiveCamera(60, dimensions.ratio, 0.001, 500000);
-            //this.camera.up.set(0, 0, 1);
-            // controls
-            this.controls = new THREE.TrackballControls(this.camera, this.renderer.domElement);
-            this.controls.rotateSpeed = 2;
-            this.controls.zoomSpeed = 2;
-            this.controls.panSpeed = 0.5;
-            this.controls.addEventListener('change', this.render.bind(this));
-            this.controls.addEventListener('start', this.controlsStart.bind(this));
-            this.controls.addEventListener('end', this.controlsEnd.bind(this));
-            // Scene and RoomEnvironment
-            const environment = new THREE.RoomEnvironment(this.renderer);
-            const pmremGenerator = new THREE.PMREMGenerator(this.renderer);
-            this.scene = new THREE.Scene();
-            this.scene.background = new THREE.Color(0xbbbbbb);
-            //this.scene.background = new THREE.Color(0x0000000);
-            this.scene.environment = pmremGenerator.fromScene(environment).texture;
-            environment.dispose();
-            TD.load('glb', model, source);
-        };
+            // Compute axis-aligned bounding box (AABB) and center (x, y, z) in
+            // world space. Size, defined here as a cheap approximation of
+            // model's maximum span in any direction, is computed from the AABB.
+            const box = new THREE.Box3().setFromObject(object);
+            const center = box.getCenter(new THREE.Vector3());
+            const size = box.getSize(new THREE.Vector3()).length();
 
+            // Center content at the origin, (0, 0, 0).
+            object.position.x -= center.x;
+            object.position.y -= center.y;
+            object.position.z -= center.z;
+
+            // In GLB, Y-Up is required. In STL it's unclear, but likely that
+            // Y-Up is typical, though Z-Up is also common. three.js controls
+            // mostly assume Y-Up. To apply a Z-Up to Y-Up conversion:
+            // object.set( - Math.PI / 2, 0, 0 );
```

Comment:
The glTF file format is very specific about its coordinate conventions:

https://registry.khronos.org/glTF/specs/2.0/glTF-2.0.html#coordinate-system-and-units

The requirement of Y-Up is pretty consistently respected and I think we can rely on that here. The "glTF asset faces +Z" line is more subjective and not strictly followed, but at least gives 3D model authors something to reference if they care about this.

For STL, there's no real convention, models can be Y-Up or Z-Up or ... who knows. I don't know of any expectation about which direction is forward. 

three.js typically uses Y-Up, so (in case we need it later) I've included a Z-Up to Y-Up conversion here, commented out.

Reply:
Hm, I see that in production Z-Up is currently used. So probably I should enable this line just for STL models, and I'll convert all of the STL samples in the repo to be Z-Up in a separate PR ... is that OK?

Reply:
Gladly. I think some people would otherwise be disappointed if their models were suddenly lying on their side. 
On the other hand, it would of course be a good opportunity to standardise this now.